### PR TITLE
Add eventSignup endpoint and new getAllReservations EventService method.

### PR DESCRIPTION
**Pull request by jeffplays2005**
Created new eventSignup endpoint and their controller.

Takes in a EventSignupBody and returns an EventSignupResponse.

Created EventService `getAllReservations` method and test.

### CODE REVIEW

**Comment by jeffplays2005** [comment]
Will create tests in an upcoming commit

File: server/src/service-layer/controllers/EventController.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      reservations.length >= fetchedEvent.max_occupancy
+    ) {
+      this.setStatus(400)
+      return { error: "Maximum event occupancy reached." }
+    }
+    // Check if the user is already signed up
+    if (
+      reservations.some(
+        (r) => r.email.toLowerCase() === reservation.email.toLowerCase()
+      )
```
Comment (bcho892) [inline server/src/service-layer/controllers/EventController.ts:37]: nit: `trim` may be good practice for this

File: server/src/service-layer/controllers/EventController.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    // Check if the event exists
+    const fetchedEvent = await eventService.getEventById(event_id)
+    if (!fetchedEvent) {
+      this.setStatus(404)
+      return { error: "Event not found." }
+    }
+    // Check if the event is full
+    const reservations = await eventService.getAllReservations(event_id)
+    if (
+      fetchedEvent.max_occupancy !== undefined &&
```
Comment (bcho892) [inline server/src/service-layer/controllers/EventController.ts:27]:
```suggestion
      !!fetchedEvent.max_occupancy &&
```

**Review by bcho892 — APPROVED** [review]
LGTM

**Pull request merged**